### PR TITLE
Make ssh port configurable in k8s deployment

### PR DESCRIPTION
**Pull request by abuccts**
Make ssh port configurable in k8s deployment.

### CODE REVIEW

File: kubernetes-deployment/cluster-config-example.yaml [hunk 1]
```diff
@@ -123,9 +123,11 @@ mastermachinelist:
     # The nodename in k8s. To avoid complication, it is suggested to fill in the hostip for the nodename.
     # If you are confident that DNS can resolve your named nodename correctly, you can specify a UNIQUE_HOST_NAME
     # hostip: the static IP of this node. We assume the IP of a host will not change.
+    # hostport: the ssh port of this node. Default port is 22.
```
Comment (fanyangCS) [inline kubernetes-deployment/cluster-config-example.yaml]: sshport seems to be a better name?

File: kubernetes-deployment/bootstrap.py [hunk 2]
```diff
@@ -96,6 +96,8 @@ def sftp_paramiko(src, dst, filename, host_config):
     username = host_config['username']
     password = host_config['password']
     port = 22
+    if (host_config['hostport']):
```
Comment (fanyangCS) [inline kubernetes-deployment/bootstrap.py]: why do we need to make ssh port in a host env configuration?

File: kubernetes-deployment/cluster-config-example.yaml (hunk 1 above)
Comment (abuccts) [reply kubernetes-deployment/cluster-config-example.yaml]: updated

File: kubernetes-deployment/bootstrap.py (hunk 2 above)
Comment (abuccts) [reply kubernetes-deployment/bootstrap.py]: some env may not allow ssh to use 22 port

File: kubernetes-deployment/cluster-config-example.yaml [hunk 3]
```diff
@@ -123,9 +123,11 @@ mastermachinelist:
     # The nodename in k8s. To avoid complication, it is suggested to fill in the hostip for the nodename.
     # If you are confident that DNS can resolve your named nodename correctly, you can specify a UNIQUE_HOST_NAME
     # hostip: the static IP of this node. We assume the IP of a host will not change.
+    # sshport: the ssh port of this node. Default port is 22.
```
Comment (fanyangCS) [inline kubernetes-deployment/cluster-config-example.yaml]: >Default [](start = 42, length = 7)

if not specified, the port is 22

**Pull request merged**